### ISSUE TITLE
Ouput Dimension

### ISSUE THREAD

**Issue by shresthamalik** [BichenWuUCB/squeezeDet#102]
What is the output dimension of self.preds (line 77)? Can anyone share?
here: 
https://github.com/BichenWuUCB/squeezeDet/blob/master/src/nets/squeezeDet.py

**Comment by supernlogn** [comment]
I have created this script long ago to do what you asked and I just uploaded it:
https://gist.github.com/supernlogn/8438c4d25443c700ac6564935372c41f#file-squeezedet_shape_calculator-py

**Comment by shresthamalik** [comment]
Thanks for sharing.

I put a print statement , here:
https://github.com/BichenWuUCB/squeezeDet/blob/master/src/nets/squeezeDet.py just after self.preds is calculated (line 77).

to find the shape of self.preds
 print (" conv 12 preds shape ",np.shape(self.preds))

I get different results for the shape above and from your script. :) May be you can check.
Shape above gives:
 conv 12 preds shape  (1, 24, 78, 72)

And your script gives:
shape  (22, 76, 72)

I am using the ./src/demo.py shared in the repo to get the values.

**Comment by shresthamalik** [comment]
In the output dimension here (1,24,78,72) , is it safe to assume that the input image is being divided into 24x78 grid and we are getting 72 outputs for each grid? I am trying to port the training code and model to caffe. So trying to make sense of how to define the loss layer.

**Comment by supernlogn** [comment]
[1,24,78,72] = [BATCH_SIZE, Hf, Wf, K *(5+C)] as figure 3 shows here: https://arxiv.org/pdf/1612.01051.pdf .
Sorry the script was actually for SqueezeDetPlus not for SqueezeDet.
But your way is generally better!
I think your problem is rewritting in caffe the _add_interpretation_graph, filter_predictions and _add_loss_graph in 
nn_skeleton.py.
